tket/src/serialize/pytket/encoder.rs

```diff
@@ -851,6 +858,70 @@ impl<H: HugrView> PytketEncoderContext<H> {
         Ok(EncodeStatus::Unsupported)
     }
 
+    /// The toposort traversal in `run_encoder` only explores nodes in the
+    /// region.
+    ///
+    /// When a node has a non-local input, we must process its originating node
+    /// before trying to translate it.
+    ///
+    /// In general if a node has a non-local dataflow input we report it as unsupported,
+    /// unless the input comes from a global definition that we are able to encode.
+    ///
+    /// # Returns
+    ///
+    /// - [`EncodeStatus::Success`] if all node inputs are supported.
+    /// - [`EncodeStatus::Unsupported`] if the node has unsupported non-local dataflow inputs, and we should mark it as unsupported.
+    fn encode_nonlocal_inputs(
+        &mut self,
+        node: H::Node,
+        optype: &OpType,
+        circ: &Circuit<H>,
+    ) -> Result<EncodeStatus, PytketEncodeError<H::Node>> {
+        let node_parent = circ.hugr().get_parent(node);
+
+        // Explore the dataflow value and static inputs, but not the _other inputs_.
+        let input_ports = circ
+            .hugr()
+            .node_inputs(node)
+            .take(optype.value_input_count() + optype.static_input_port().is_some() as usize);
```

Comment:
I find this quite ugly and have thought multiple times that we should have a better API for specifying the kind of ports that we want to iterate over.

Maybe `node_ports_of_kind([Kind::ValuePort, Kind::StaticPort])`? But not for this PR :)